profiling/src/profile/mod.rs

```diff
@@ -555,65 +350,83 @@ impl Profile {
     }
 
     /// Interns the `str` as a string, returning the id in the string table.
-    fn intern(&mut self, str: &str) -> StringId {
-        // strings are special because the empty string is actually allowed at
-        // index 0; most other 0's are reserved and cannot exist
-        StringId::new(self.strings.dedup_ref(str))
+    fn intern(&mut self, item: &str) -> StringId {
```

Comment:
I was going to ask why we need this given that its a repeat of the `dedup` logic. But I assume this is because we don't want to have to convert `&str` to `String` for the `insert_full`?

Reply:
Yes, we don't want to convert `&str` into `String` unless we have to -- that's a good chunk of overhead that we can avoid for cases where it already exists in the set.

Reply:
(Should we add this as a comment on the code? Seems like a question that will come up when others look at this code :P )

Reply:
I added some commentary in the code 👍🏻